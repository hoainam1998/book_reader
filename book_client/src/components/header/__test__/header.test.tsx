import Header from '../header';
import { render, screen } from '@testing-library/react';
import { createBrowserRouter, createRoutesFromElements, Route, RouterProvider } from 'react-router-dom';

jest.mock('storage', () => {
  return {
    __esModule: true,
    UserStorage: {
      getItem: jest.fn(() => ({
        name: 'name',
        email: 'email@gmail.com',
        avatar: 'avatar',
        mfaEnable: true,
        password: 'namkho@19'
      }))
    }
  };
});

describe('Header', () => {
  it('header should render correctly', () => {
    const router = createBrowserRouter(
      createRoutesFromElements(<Route path="/" element={<Header />} />)
    );
    render(<RouterProvider router={router} />);
    const name = screen.getByTestId('name');
    const email = screen.getByTestId('email');
    expect(name.textContent).toBe('name');
    expect(email.textContent).toBe('email@gmail.com');
  });
});
