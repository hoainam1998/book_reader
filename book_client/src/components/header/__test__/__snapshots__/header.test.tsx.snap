// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header header should render correctly 1`] = `
<header
  className="header"
>
  <img
    alt="logo"
    height="40"
    src="book.png"
    width="40"
  />
  <div
    className="personal-information-group"
  >
    <button
      className="button button-avatar"
      onClick={[Function]}
    >
      <img
        alt="logo"
        height="30"
        src="book.png"
        width="30"
      />
    </button>
    <span>
      username
    </span>
  </div>
</header>
`;
